pulsar/producer_partition.go

```diff
@@ -553,9 +613,7 @@ func (p *partitionProducer) ReceivedSendReceipt(response *pb.CommandSendReceipt)
 	// lock the pending item while sending the requests
 	pi.Lock()
 	defer pi.Unlock()
-	if pi.sentAt > 0 {
```

Comment:
@cckellogg 
`sentAt` always set before puting into pending queue, so I removed this predicate.
Like [sendRequest.publishTime](https://github.com/apache/pulsar-client-go/pull/394/files#diff-8adb09af2175be5751840343e14df2220a2154ca4bc8de21157ed112be25b6b9L565) did